csfieldguide/chapters/content/en/coding-compression/sections/general-purpose.md

```diff
@@ -43,27 +31,337 @@ Here are some of the bits from the example earlier in this chapter; you can past
+| 7    | 000000000111                 | SA       |
+| 8    | 000000001000                 | AMS      |
+| 9    | 000000001001                 | SAM      |
+| 10   | 000000001010                 | MI       |
+| 11   | 000000001011                 | IAM      |
+
+And the string has been replaced with a series of codes, each corresponding to a value in the dictionary.
+`0 1 2 3 5 7 2 4 2`
+
+To see just how much compression we achieved, let’s take a look at the binary behind the characters.
+We'll assume that the characters are coded in 8 bits each (this is typical), based on UTF-8 coding (which is the same as ASCII for English characters).
+In binary, originally our message looked like this:
+
+```
+IAMSAMSAMIAM
+
+01001001 01000001 01001101 01010011 01000001 01001101 01010011 01000001 01001101 01001001 01000001 01001101
+```
+
+The uncompressed version has 12 characters, each using 8 bits, so there are 92 bits used to store it.
+
+And this is the compressed message:
+```
+000000000000 000000000001 000000000010 000000000011 000000000101 000000000111 000000000010 000000000100 000000000010
+```
+
+The compressed version shown here uses only 9 LZW codes, but each is 12 bits each, so this version would use 108 bits, which is more than the original, although we'll see later on (check out the "Extra for experts: fewer than 12 bits" panel at the end of this section) that there's an easy trick that can be used to reduce this to just 3 or 4 bits per code, so it would use no more than 4x9, or 36 bits, which is less than 40% of the original size.
+
+We were encoding quite a short message, so the longest sequence in the dictionary is just three characters.
+The longer the message that we are encoding, the longer these sequences can get, therefore allowing us to potentially represent reasonably long substrings with just one code, resulting in effective compression.
+You can see this for yourself in the LZW compression interactive below:
+
+{interactive slug="lzw-compression" parameters="message-length=long" type="whole-page"}
+
+{panel type="challenge"}
+
+# Challenge
+
+This compression algorithm works best for text with plenty of repeated sequences.
+To see this for yourself, try encoding the following string using the LZW interactive:
+
+```
+The fox jumped over the lazy dog.
+```
+
+While this string is longer than the one used in our example, there are fewer repeated sequences and therefore not nearly as much opportunity for compression!
+
+{panel end}
+
+
+## LZW Decoding
+
+Decoding a message from LZW is a little tricky, but makes an interesting puzzle.
+The main point is that anything that has been encoded can be decoded to exactly the same as it was before it was compressed, but here's an illustration of how it works in case you're curious about how it's even possible.
+Let’s now try to decode our encoded message.
+
+The decoder starts with the same initial dictionary of single characters as the encoder:
```

Comment:
Explain _how_ the decoder gets the same initial dictionary of single characters as the encoder. The only way I can think of it being done is for the initial dict to be sent as data - but later on you say that is not the case